src/client/common/installer/backupPipInstaller.ts

```diff
@@ -10,22 +10,37 @@ import { IApplicationShell, IWorkspaceService } from '../application/types';
 import { wrapCancellationTokens } from '../cancellation';
 import { STANDARD_OUTPUT_CHANNEL } from '../constants';
 import { disposeAllDisposables } from '../helpers';
-import { traceError, traceVerbose, traceWarning } from '../logger';
+import { traceError, traceVerbose } from '../logger';
 import { getDisplayPath } from '../platform/fs-paths';
 import { IPythonExecutionFactory } from '../process/types';
 import { IDisposable, IOutputChannel, Product, Resource } from '../types';
 import { DataScience } from '../utils/localize';
 import { translateProductToModule } from './productInstaller';
 import { ProductNames } from './productNames';
 
+function isGlobalEnvironment(interpreter: PythonEnvironment): boolean {
+    return (
+        interpreter.envType === EnvironmentType.Global ||
+        interpreter.envType === EnvironmentType.System ||
+        interpreter.envType === EnvironmentType.WindowsStore
+    );
+}
+function isVirtualEnv(interpreter: PythonEnvironment): boolean {
+    return (
+        interpreter.envType === EnvironmentType.Venv ||
+        interpreter.envType === EnvironmentType.VirtualEnv ||
+        interpreter.envType === EnvironmentType.VirtualEnvWrapper
+    );
+}
 /**
- * Class used to install IPyKernel into global enviroments if Python extension fails to install it for what ever reason.
+ * Class used to install IPyKernel into global environments if Python extension fails to install it for what ever reason.
```

Comment:
If you notice, I've fixed a typo (installed spell checker, for some reason this was uninstalled on my mac)